service-workers/service-worker/fetch-event-async-respond-with.https.html

```diff
@@ -3,31 +3,33 @@
 <script src="resources/testharness-helpers.js"></script>
 <script src="/resources/testharnessreport.js"></script>
 <script src="resources/test-helpers.sub.js"></script>
+<script src="resources/service-worker-recorder.js"></script>
 <script>
 promise_test(function(t) {
     var script = 'resources/fetch-event-async-respond-with-worker.js';
     var scope = 'resources/simple.html';
 
     return service_worker_unregister_and_register(t, script, scope)
       .then(function(registration) {
-          return wait_for_state(t, registration.installing, 'activated');
+          return wait_for_state(t, registration.installing, 'activated')
+            .then(function() {
```

Comment:
Why the nested promise chains here (and in some other places)? You could just list this then up to the main promise chain. (i.e. register.then(wait for state).then(clear).then(with iframe)...)